modules/auxiliary/scanner/rsync/modules_list.rb

```diff
@@ -24,46 +23,97 @@ def initialize
     )
     register_options(
       [
-        Opt::RPORT(873)
+        Opt::RPORT(873),
+        OptBool.new('AUTH_CHECK', [true, 'Check authentication or not', false])
       ], self.class)
   end
 
-  def run_host(ip)
+  def rsync(dir)
     connect
-    version = sock.get_once
 
+    version = sock.get_once # server_initialisation
     return if version.blank?
 
-    print_good("#{ip}:#{rport} - rsync #{version.strip} found")
-    report_service(:host => ip, :port => rport, :proto => 'tcp', :name => 'rsync')
-    report_note(
-        :host => ip,
-        :proto => 'tcp',
-        :port => rport,
-        :type => 'rsync_version',
-        :data => version.strip
-    )
-
-    # making sure we match the version of the server
-    sock.puts("#{version}")
-    # the listing command
-    sock.puts("\n")
-    listing = sock.get(20)
+    sock.get(3) # server_motd
+    sock.puts(version) # client_initialisation
```

Comment:
This should check to see that the data we got back looks like rsync, otherwise this module will falsely report /interact like this is rsync when it isn't, like https://github.com/jhart-r7/metasploit-framework/blob/feature/rsync-module-list-cleanup/modules/auxiliary/scanner/rsync/modules_list.rb#L100-L103